packages/expo-router/src/rsc/rsc-renderer.ts

```diff
@@ -0,0 +1,264 @@
+
+import { runWithRenderStore, type EntriesDev, type EntriesPrd } from './server';
+import { getServerReference, getDebugDescription } from '../server-actions';
+
+// Make global so we only pull in one instance for state saved in the react-server-dom-webpack package.
+// @ts-ignore: HACK type for server actions
+globalThis._REACT_registerServerReference = registerServerReference;
+
+export interface RenderContext<T = unknown> {
+  rerender: (input: string, searchParams?: URLSearchParams) => void;
+  context: T;
+}
+
+type ResolvedConfig = any;
+
+export type RenderRscArgs = {
+  // TODO:
+  config: ResolvedConfig;
+
+  // Done
+  input: string;
+  searchParams: URLSearchParams;
+  method: 'GET' | 'POST';
+  context: Record<string, unknown> | undefined;
+  body?: ReadableStream | undefined;
+  contentType?: string | undefined;
+  moduleIdCallback?: (module: {
+    id: string;
+    chunks: string[];
+    name: string;
+    async: boolean;
+  }) => void;
+};
+
+type ResolveClientEntry = (id: string) => { id: string; chunks: string[] };
+
+type RenderRscOpts =
+  | {
+      isExporting: true;
+      entries: EntriesPrd;
+      resolveClientEntry?: ResolveClientEntry;
+    }
+  | {
+      isExporting: false;
+      entries: EntriesDev;
+      resolveClientEntry: ResolveClientEntry;
+    };
+
+export async function renderRsc(args: RenderRscArgs, opts: RenderRscOpts): Promise<ReadableStream> {
+  const { searchParams, method, input, body, contentType, context } = args;
+  const { resolveClientEntry, entries } = opts;
+
+  const {
+    default: { renderEntries },
+    // @ts-expect-error
+    buildConfig,
+  } = entries as (EntriesDev & { loadModule: never; buildConfig: never }) | EntriesPrd;
+
+  const bundlerConfig = new Proxy(
+    {},
+    {
+      get(_target, encodedId: string) {
+        const [
+          // File is the on-disk location of the module, this is injected during the "use client" transformation (babel).
+          file,
+          // The name of the import (e.g. "default" or "")
+          // This will be empty when using `module.exports = ` and `require('...')`.
+          name = '',
+        ] = encodedId.split('#') as [string, string];
+
+        // HACK: Special handling for server actions being recursively resolved, e.g. ai demo.
+        if (encodedId.match(/[0-9a-z]{40}#/i)) {
+          return { id: encodedId, chunks: [encodedId], name, async: true };
+        }
+
+        const filePath = file.startsWith('file://') ? fileURLToFilePath(file) : file;
+
+        // We'll augment the file path with the incoming RSC request which will forward the metro props required to make a cache hit, e.g. platform=web&...
+        // This is similar to how we handle lazy bundling.
+        if (resolveClientEntry) {
+          const resolved = resolveClientEntry(filePath);
+          return { id: resolved.id, chunks: resolved.chunks, name, async: true };
+        }
+
+        return {
+          // TODO: Make relative to server root
```

Comment:
I think this one might be important, especially when we enable `EXPO_USE_METRO_WORKSPACE_ROOT` by default.

Reply:
for now we use absolute paths. This code path is production-only which won't be supported until the next PR.